### ISSUE TITLE
Introduce exponential backoff when IDE reconnects with backend

### ISSUE THREAD

**Issue by wdanilo** [enso-org/enso#5066]
This task is automatically imported from the old Task Issue Board and it was originally created by Hubert Plociniczak.
Original issue is [here](https://www.pivotaltracker.com/story/show/184358480).
<hr />

Currently if we kill backend (LS), IDE will keep on trying to reconnect at the same rate which is not particularly useful. Neither is the message `Connection to the backend has been lost. Please try restarting IDE`.

We should:
- apply gmail strategy which uses exponential backoff to reconnect in case of failure
- update the message with the information on when IDE would attempt to reconnect
- preferably provide a more visible error message at the top (gray on white is easy to miss)

Discussion thread: https://discord.com/channels/401396655599124480/1069912865261240371

**Comment by Frizi** [comment]
Depends on #5202. Once the reconnection procedure itself is implemented, adding exponential backoff should be purely IDE side modification.

**Comment by farmaazon** [comment]
Superseded by #8520
